sources/main.cpp

```diff
@@ -209,23 +209,36 @@ int main(int argc, char* argv[]) {
 
     // reset combination was pressed on startup
     if (buttonHandler->resetButtonsPressed()) {
-        // create marker file
-        QFile file("/firstrun");
-        if (file.open(QIODevice::WriteOnly)) {
-            file.close();
-        }
-
-        // restore default config
-        if (QFile::exists("/boot/config.json")) {
-            QFile::remove("/boot/config.json");
+        if (QFile::exists(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json") &&
+            QFile::exists("/boot/config.json")) {
+            // create marker file
+            QFile file("/firstrun");
+            if (file.open(QIODevice::WriteOnly)) {
+                file.close();
+
+                // rename existing config
+                QFile::rename("/boot/config.json", "/boot/config.json.old");
+
+                // copy default config
+                QFile::copy(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json",
+                            "/boot/config.json");
```

Comment:
Check return value if copy operation failed, otherwise the remote will use the default config in the app directory after reboot.
If copy failed:
- Log error
- Restore from `/boot/config.json.old`